doc/conf.py

```diff
@@ -1000,7 +885,32 @@ def setup(app):
         "https://github.com/": {"Authorization": f"token {github_token}"},
     }
 
-# Write the pages prior to any sphinx event
-generate_index_rst()
-generate_min_dependency_table()
-generate_min_dependency_substitutions()
+# Convert the template rst files into actual rst files prior to any sphinx event
+
+from sklearn._min_dependencies import dependent_packages
+
+# (template name, file name, kwargs for rendering)
+rst_templates = [
+    ("index", "index", {"is_devrelease": parsed_version.is_devrelease}),
```

Comment:
Nit: Can we do the `development_link` logic here instead of in the Jinja template?

```suggestion
if parsed_version.is_devrelease:
    development_link = "developers/index"
else:
    development_link = "https://scikit-learn.org/dev/developers/index.html"

# (template name, file name, kwargs for rendering)
rst_templates = [
    ("index", "index", {"development_link": development_link}),
```

(If we can get away with it, I prefer less logic in the Jinja template)

Reply:
Can't agree more. I've moved the logic into `conf.py`.